### ISSUE TITLE
Admin Color Scheme styles leaking into the Site Editor

### ISSUE THREAD

**Issue by alaczek** [Automattic/wp-calypso#86778]
### Quick summary

In the site editor, selecting the paragraph or text block turns it pink, which depending on the global styles can make the text unreadable due to low contrast. This seems to only happen on WordPress.com simple sites.

### Steps to reproduce

1. On a simple site with a block theme active, go to Appearance > Editor.
2. Click on Templates and select any of the templates to edit.
3. Select or create a text block (paragraph, heading, navigation, site title, etc.). Notice that it the text turns pink when the block is selected.
4. In the global styles settings change the site background color to #DA1035. Notice the text is unreadable now when a paragraph/heading block is selected.

<img width="584" alt="image" src="https://github.com/Automattic/wp-calypso/assets/4550351/ec06096e-96fa-4bad-9300-406c66957168">

### What you expected to happen

I expected for the text not to change the color at all when a text-based block is selected. This aligns with core behavior, and avoids the issue of text potentially clashing with global styles and being unreadable while editing.

### What actually happened

Selecting a text-based block made it pink.

### Impact

Most (> 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

This seems to only happen on WordPress.com simple sites.
I was not able to replicate on a local install with WP 6.4.2 and Gutenberg 17.5.1 or on an Atomic site with Gutenberg 17.5.1

**Comment by liviopv** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
When I select text on a Simple Site in the site editor using TT4, a bluish background is applied to the selected text, in a hue that I believe is derived from the #3858e9 applied to the admin color I defined in `/me/account`. It's a little different than core, but it doesn't seem to cause any accessibility issue

Local Site X Simple Site comparison
![CleanShot 2024-01-24 at 16 07 25@2x](https://github.com/Automattic/wp-calypso/assets/14153300/93f62160-f268-4042-86a7-7db77f7f186a)


📌 ACTIONS
- Requested author feedback


📌 Message to Author
@alaczek I feel like I'm missing something here, as I'm not able to reproduce - or perhaps I didn't test the correct thing. Is it theme-specific? Or perhaps only reproducible when selecting a specific dashboard color scheme on https://wordpress.com/me/account?

**Comment by alaczek** [comment]
I was able to reproduce this with a few themes (TT4, Bedrock, Smithland). You're right that it feels like it's related to the admin color scheme (Interface Settings in user profile). I switched from Bright to Default, and the issue persists. It's not about the bluish highlight though - it's about the text turning pink - I made a quick video:

https://github.com/Automattic/wp-calypso/assets/4550351/3c371775-d107-4e4a-a542-daf2bf15082e

**Comment by alaczek** [comment]
It's definitely related to the interface color scheme in user profile - the text turns orange when selected if I pick the Sunset color scheme and teal when I picked Ocean color scheme. And in the Default and Bright the selected text is pink, because these two have the same accent color.
<img width="645" alt="image" src="https://github.com/Automattic/wp-calypso/assets/4550351/8a79e217-958b-4731-bd71-798d70f37cbb">



Possibly it's this piece of CSS leaking into the editor:
![image](https://github.com/Automattic/wp-calypso/assets/4550351/e1cd83b0-da87-4cf5-a421-021228d83fb6)

**Comment by liviopv** [comment]
I was able to reproduce when switching to the Ocean color scheme - select text inside the editor changes color, which can cause an accessibility issue depending on the background:

![CleanShot 2024-01-25 at 11 20 31@2x](https://github.com/Automattic/wp-calypso/assets/14153300/2114cc7c-aa88-491e-83ed-5ea28c475de6)

**Comment by miksansegundo** [comment]
We fixed a similar CSS issue related to the admin styles leaking into the editor canvas by disabling the concatenation of that asset. 

The other issue was caused by the [reset.css](https://github.com/WordPress/gutenberg/blob/940f0fe52cfd1e3599e507a7d3f83146914e978f/packages/block-library/src/reset.scss#L89) affecting the margin styles of the Heading block. See D134512-code for Gutenberg: Fix for the Heading block styles on h2 and h3 avoiding the concatenation of the editor reset.css.

**Comment by miksansegundo** [comment]
I cannot reproduce it on TT4, or Bedrock, using Classic bright and dark admin color schemes.

@alaczek, please let us know if you can reproduce this or any other issues related to wp-admin styles leaking into the editor canvas because we know how to fix them.

<img width="297" alt="Screenshot 2567-05-20 at 11 36 07" src="https://github.com/Automattic/wp-calypso/assets/1881481/67146730-72ef-4de7-b091-4506beb5eefa">

<img width="1728" alt="Screenshot 2567-05-20 at 11 38 50" src="https://github.com/Automattic/wp-calypso/assets/1881481/634848cc-28a4-4e7c-a86f-f59a6689b1b8">

<img width="1728" alt="Screenshot 2567-05-20 at 11 36 43" src="https://github.com/Automattic/wp-calypso/assets/1881481/d4d73933-c13e-4dc0-9d3a-d017da8b3e84">
